File: serialize.py
def serialize(obj):
    # String
    if isinstance(obj, str):
        return f'"{obj}"'

    # Integer or Float
    elif isinstance(obj, (int, float)):
        return str(obj)

    # Boolean
    elif isinstance(obj, bool):
        return 'true' if obj else 'false'

    # None se null karna h
    elif obj is None:
        return 'null'

    # Dictionary
    elif isinstance(obj, dict):
        items = [f"{serialize(key)}:{serialize(value)}" for key, value in obj.items()]
        return '{' + ','.join(items) + '}'

    # Lists
    elif isinstance(obj, list):
        items = [serialize(item) for item in obj]
        return '[' + ','.join(items) + ']'

    else:
        raise TypeError(f'Unsupported type: {type(obj)}')

File: serialize.go
package main

import (
	"fmt"
	"strings"
)

// Serialize converts a Go value into a JSON string.
func Serialize(value interface{}) string {
	switch v := value.(type) {

	// Handle string values
	case string:
		return `"` + v + `"`

	// Handle numeric values
	case int, float64:
		return fmt.Sprintf("%v", v)

	// Handle boolean values
	case bool:
		if v {
			return "true"
		}
		return "false"

	// Handle null value
	case nil:
		return "null"

	// Handle arrays
	case []interface{}:
		var items []string
		for _, item := range v {
			items = append(items, Serialize(item))
		}
		return "[" + strings.Join(items, ",") + "]"

	// Handle objects
	case map[string]interface{}:
		var items []string
		for key, val := range v {
			items = append(items, Serialize(key)+":"+Serialize(val))
		}
		return "{" + strings.Join(items, ",") + "}"

	// Handle unsupported types
	default:
		panic(fmt.Sprintf("unsupported type: %T", v))
	}
}
